### ISSUE TITLE
[Java 1.7] MQTT runtime exception due to paho 1.2.2

### ISSUE THREAD

**Issue by timtay-microsoft** [Azure/azure-iot-sdk-java#596]
Details on the paho issue [here](https://github.com/eclipse/paho.mqtt.java/issues/633)

The latest release of the java device client/deps packages takes a dependency on paho 1.2.2 which encounters a runtime exception while opening a connection using mqtt.

This exception seems to be limited to mqtt users below Android 7, as well as Java SDK version 1.7 and below. JDK 1.8+ works as expected.

**Comment by timtay-microsoft** [comment]
Android API levels lower than 24 are affected

**Comment by GiovanniVisentini** [comment]
As workarround I add this lines to my main project gradle to force version 1.20 that is compatible with lower android api. 

```Gradle
components.all {
    allVariants {
        withDependencies { deps ->
            deps.each { dep ->
                if (dep.group == 'org.eclipse.paho' && dep.name =='org.eclipse.paho.client.mqttv3') {
                    dep.version {
                        require "1.2.0"
                     }
                     dep.because "require api 24: https://github.com/eclipse/paho.mqtt.java/issues/633"
                }
            }
        }
    }
}
```

**Comment by timtay-microsoft** [comment]
The above will work fine for most people. This SDK upgraded the paho dependency up to 1.2.2 to get proxy support features, so be aware that using 1.2.0 will break proxy support over mqtt

**Comment by timtay-microsoft** [comment]
Just as an update here, the fix for the Paho issue has been checked into the Paho repo, so now this SDK is just waiting on the next release of the Paho library

**Comment by timtay-microsoft** [comment]
I have tried out the new version of Paho (1.2.3), and I'm still seeing some errors when this sdk tries to connect for Java 7 or Android API level 24 and lower. It doesn't appear that we can use this new version to fix this issue. I'll work with the Paho devs to understand this new issue

**Comment by nazmul32** [comment]
@timtay-microsoft 

I believe this issue exists in azure IoT SDK 1.27.0. It is crashing due to **SNIHostName** issue when connecting with Android 6 devices. 

01-07 02:06:59.761 30342-30500/? E/AndroidRuntime: FATAL EXCEPTION: MQTT Con: fd7da83c-f2d9-4979-9801-531f32c70f55
Process: com.shohoz.deliveryagent, PID: 30342
java.lang.NoClassDefFoundError: Failed resolution of: Ljavax/net/ssl/SNIHostName;
at org.eclipse.paho.client.mqttv3.internal.SSLNetworkModule.start(SSLNetworkModule.java:139)
at org.eclipse.paho.client.mqttv3.internal.ClientComms$ConnectBG.run(ClientComms.java:722)
at java.lang.Thread.run(Thread.java:818)
Caused by: java.lang.ClassNotFoundException: Didn't find class "javax.net.ssl.SNIHostName" on path: DexPathList[[zip file "/data/app/com.shohoz.deliveryagent-1/base.apk"],nativeLibraryDirectories=[/data/app/com.shohoz.deliveryagent-1/lib/arm, /vendor/lib, /system/lib]]
at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
at java.lang.ClassLoader.loadClass(ClassLoader.java:511)
at java.lang.ClassLoader.loadClass(ClassLoader.java:469)
at org.eclipse.paho.client.mqttv3.internal.SSLNetworkModule.start(SSLNetworkModule.java:139) 
at org.eclipse.paho.client.mqttv3.internal.ClientComms$ConnectBG.run(ClientComms.java:722) 
at java.lang.Thread.run(Thread.java:818) 
Suppressed: java.lang.ClassNotFoundException: javax.net.ssl.SNIHostName
at java.lang.Class.classForName(Native Method)
at java.lang.BootClassLoader.findClass(ClassLoader.java:781)
at java.lang.BootClassLoader.loadClass(ClassLoader.java:841)
at java.lang.ClassLoader.loadClass(ClassLoader.java:504)
... 4 more
Caused by: java.lang.NoClassDefFoundError: Class not found using the boot class loader; no stack trace available
01-07 02:06:59.811 726-30511/? E/android.os.Debug: ro.product_ship = true``

**Comment by timtay-microsoft** [comment]
Closing this issue as "won't fix" because the device client library no longer supports Java 7 or below

**Comment by az-iot-builder-01** [comment]
@timtay-microsoft, @GiovanniVisentiniST, @nazmul32, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)
